pkg/frontend/admin_hive_clusterdeployment_get.go

```diff
@@ -0,0 +1,63 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) getAdminHiveClusterDeployment(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	b, err := f._getAdminHiveClusterDeployment(ctx, r)
+
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeNotFound, "", "Cluster not found.")
+		return
+	case err != nil:
+		api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", err.Error())
+		return
+	}
```

Comment:
It might be worth moving this logic into `_getAdminHiveClusterDeployment()` at the moment that you Get the cluster document (line 42)

Reply:
I agree, the error handling can be simplified. I'll get this refactoring in.

Reply:
changes done!